cmd/commands/runtime_install.go

```diff
@@ -2212,3 +2172,184 @@ func initializeGitSourceCloneOpts(opts *RuntimeInstallOptions) {
 	host, orgRepo, _, _, _, suffix, _ := aputil.ParseGitUrl(opts.InsCloneOpts.Repo)
 	opts.GsCloneOpts.Repo = host + orgRepo + "_git-source" + suffix + "/resources" + "_" + opts.RuntimeName
 }
+
+// bitbucket cloud take more time to push a commit
+// all coming retries perpuse is to avoid issues of cloning before pervious commit was pushed
+
+func persistOrUpdateRuntime(ctx context.Context, opts *RuntimeInstallOptions, rt *runtime.Runtime, server string) error {
+
+	var err error
+	for try := 0; try < pushRetries; try++ {
```

Comment:
maybe we can make some `func Retry` in utils, that accepts a func, and just retries it.
it can have some extra optional parameters, like total number of retries, and optional sleep time, with sensible defaults.